### PR TITLE
Fixes/cleanup for Unit.unitid

### PR DESCRIPTION
**Pull request by phlax**
and adds test to ensure that unit is created correctly

### CODE REVIEW

File: pootle/apps/pootle_store/models.py [hunk 1]
```diff
@@ -359,7 +359,7 @@ def save(self, *args, **kwargs):
         super(Unit, self).save(*args, **kwargs)
         if created:
             unit_source = self.unit_source.model(unit=self)
-            unit_source.created_by = user
+            unit_source.created_by = submitted_by or user
```
Comment (unho) [inline pootle/apps/pootle_store/models.py]: Do you mean `self.submitted_by`?

Comment (unho) [inline pootle/apps/pootle_store/models.py]: I see it is added on a later commit.

File: tests/pootle_store/utils.py [hunk 2]
```diff
@@ -65,6 +109,7 @@ def test_unit_lifecycle_update_state(store0, member):
     unit.index = store0.max_index() + 1
     unit.reviewed_by = member
     unit.save()
+
```
Comment (unho) [inline tests/pootle_store/utils.py]: Nit: Is this blank line necessary?

File: tests/pootle_store/utils.py [hunk 3]
```diff
@@ -54,6 +94,10 @@ def test_unit_lifecycle_create(store0):
     assert source.created_by == get_system_user()
     assert source.created_with == SubmissionTypes.SYSTEM
 
+    with pytest.raises(UnitChange.DoesNotExist):
+        unit.change
+
+
```
Comment (unho) [inline tests/pootle_store/utils.py]: Three blank lines?

Comment (unho) [inline tests/pootle_store/utils.py]: Same above.

File: pootle/apps/pootle_store/models.py (hunk 1 above)
Comment (phlax) [inline pootle/apps/pootle_store/models.py]: reordered commits

File: tests/pootle_store/utils.py (hunk 3 above)
Comment (phlax) [inline tests/pootle_store/utils.py]: this is the kinda thing that tests pick up

**Comment by unho** [comment]
lgtm provided tests pass.

**Pull request merged**